### PR TITLE
Http server: Allow to disable not-found on missing body

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: core-reactive/src/main/java/io/micronaut/core/async/publisher/Publishers.java [hunk 1]
```diff
@@ -516,6 +515,27 @@ public static <T> T convertPublisher(ConversionService conversionService, Object
...
+    /**
+     * Attempts to convert the publisher to the given type.
+     *
+     * @param conversionService The conversion service
+     * @param object The object to convert
+     * @param <T> The generic type
+     * @return The Resulting in publisher
+     * @since 4.6.0
+     */
+    public static <T> Publisher<T> convertToPublisher(ConversionService conversionService, Object object) {
```
Comment (graemerocher) [inline core-reactive/src/main/java/io/micronaut/core/async/publisher/Publishers.java]: add nullability annotations

File: http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java [hunk 2]
```diff
@@ -582,6 +583,22 @@ public boolean isValidateUrl() {
         return validateUrl;
     }
 
+    /**
+     * @return True if not-found should be returned on missing body
```
Comment (graemerocher) [inline http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java]: document what happens if set to `false`

File: test-suite-http-server-tck-netty/src/test/java/io/micronaut/http/server/tck/netty/tests/NettyHttpServerTestSuite.java [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
-import org.junit.platform.suite.api.SelectPackages;
+import io.micronaut.http.server.tck.tests.NoBodyResponseTest;
+import org.junit.platform.suite.api.SelectClasses;
 import org.junit.platform.suite.api.Suite;
 import org.junit.platform.suite.api.SuiteDisplayName;
 
 @Suite
-@SelectPackages("io.micronaut.http.server.tck.tests")
+//@SelectPackages("io.micronaut.http.server.tck.tests")
+@SelectClasses(NoBodyResponseTest.class)
```
Comment (graemerocher) [inline test-suite-http-server-tck-netty/src/test/java/io/micronaut/http/server/tck/netty/tests/NettyHttpServerTestSuite.java]: revert this change

**Review by graemerocher — CHANGES_REQUESTED** [review]

**Review by sdelamo — CHANGES_REQUESTED** [review]
Please, add a mention in https://github.com/micronaut-projects/micronaut-core/blob/4.6.x/src/main/docs/guide/httpServer/serverIO.adoc#404-responses to tell users how to disable this behaviour and what would be the behaviour if they set `an `not-found-on-missing-body` to `false`

File: http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java [hunk 4]
```diff
@@ -433,13 +428,17 @@ public Runnable instrument(Runnable command) {
...
     }
 
-    private ExecutionFlow<MutableHttpResponse<?>> fromImperativeExecute(PropagatedContext propagatedContext, HttpRequest<?> request, RouteInfo<?> routeInfo, Object body) {
+    private ExecutionFlow<MutableHttpResponse<?>> fromImperativeExecute(PropagatedContext propagatedContext,
+                                                                        HttpRequest<?> request,
+                                                                        RouteInfo<?> routeInfo,
+                                                                        @Nullable Object body) {
         // performance optimization: check for common body types
-        boolean shortCircuit = body instanceof String || body instanceof byte[];
+        boolean shortCircuit = body instanceof String || body instanceof byte[] || body == null;
```
Comment (yawkat) [inline http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java]:
```suggestion
        boolean shortCircuit = body instanceof String || body instanceof byte[];
```

This change is not needed, the shortCircuit is specifically for type pollution which null isn't affected by

File: http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java [hunk 5]
```diff
@@ -678,8 +687,7 @@ private CorePublisher<MutableHttpResponse<?>> fromReactiveExecute(PropagatedCont
             }
             return response.contextWrite(context -> ReactorPropagation.addPropagatedContext(context, propagatedContext).put(ServerRequestContext.KEY, request));
         }
-        MutableHttpResponse<?> response = forStatus(routeInfo, null).body(body);
-        return processPublisherBody(propagatedContext, request, response, routeInfo);
+        return processPublisherBody(propagatedContext, request, forStatus(routeInfo, null), false, publisher, routeInfo);
```
Comment (yawkat) [inline http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java:690]: why is the `.body` gone?

**Review by yawkat — APPROVED** [review]

File: http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java (hunk 5 above)
Comment (dstepanov) [reply http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java:690]: I did a bit of refactoring to pass the converted publisher right away

**Comment by dstepanov** [comment]
Added docs @sdelamo

**Review by graemerocher — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]
